### PR TITLE
Bugfix for CancellationTokenSource

### PR DESCRIPTION
**Pull request by cltshivash**
Added initialization that was missing in the NetCoreTestContext and added a test to cover the net core scenario as well.

### CODE REVIEW

File: test/E2ETests/TestAssets/TimeoutTestProjectNetCore/TimeoutTestProjectNetCore.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TestFxRoot Condition="$(TestFxRoot) == ''">..\..\..\..\</TestFxRoot>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFramework>net452</TargetFramework>
+    <IsPackable>false</IsPackable>
```
Comment (ankit-nitd-zz) [inline test/E2ETests/TestAssets/TimeoutTestProjectNetCore/TimeoutTestProjectNetCore.csproj:9]: Shouldn't this project have target framework of net core type?

Comment (cltshivash) [reply test/E2ETests/TestAssets/TimeoutTestProjectNetCore/TimeoutTestProjectNetCore.csproj:9]: Thanks for catching this. Fixed it.

File: test/E2ETests/TestAssets/TimeoutTestProjectNetCore/TimeoutTestProjectNetCore.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TestFxRoot Condition="$(TestFxRoot) == ''">..\..\..\..\</TestFxRoot>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+    <FrameworkIdentifier>NetCore</FrameworkIdentifier>
+    <IsPackable>false</IsPackable>
+    <!--<AppendTargetFrameworkToOutputPath>true</AppendTargetFrameworkToOutputPath>-->
```
Comment (jayaranigarg) [inline test/E2ETests/TestAssets/TimeoutTestProjectNetCore/TimeoutTestProjectNetCore.csproj:10]: If we make this as false, then we will not need to specify 'netcoreapp1.1' in TimeoutTestAssemblyNetCore path here:https://github.com/Microsoft/testfx/blob/37b1ebf822786771fbcf641cfa84a39f3c83204b/test/E2ETests/Smoke.E2E.Tests/TimeoutTests.cs#L14

Comment (cltshivash) [reply test/E2ETests/TestAssets/TimeoutTestProjectNetCore/TimeoutTestProjectNetCore.csproj:10]: Thought about it. But it is better to have a separate folder per test assembly.

**Pull request merged**